src/main/java/org/fundacionjala/coding/rodrigo/OcrBank.java

```diff
@@ -0,0 +1,88 @@
+package org.fundacionjala.coding.rodrigo;
+
+/**
+ * KataBankOCR.
+ */
+public final class OcrBank {
+    static final String[] DIGITS = {
+            " _ | ||_|",    //zero
+            "     |  |",    //one
+            " _  _||_ ",    //two
+            " _  _| _|",    //three
+            "   |_|  |",    //four
+            " _ |_  _|",    //five
+            " _ |_ |_|",    //six
+            " _   |  |",    //seven
+            " _ |_||_|",    //eight
+            " _ |_| _|"};   //nine
+
+    /**
+     * Empty Constructor.
+     */
+    private OcrBank() {
+    }
+
+    /**
+     * @param account input like 7 segment.
+     * @return processed digits.
+     */
+    public static String convertOcrToNumber(final String account) {
+        final int numsep = 3;
+        String[] line = account.split("\n");
+        StringBuilder number = new StringBuilder();
+        for (int i = 0; i < line[0].length() / numsep; i++) {
+            StringBuilder aux = new StringBuilder();
+            aux.append(line[0], numsep * i, numsep + (numsep * i))
+                    .append(line[1], numsep * i, numsep + (numsep * i))
+                    .append(line[2], numsep * i, numsep + (numsep * i));
+            number.append(searchDigit(aux.toString()));
+        }
+
+        return number.toString();
+    }
+
+    /**
+     * @param account String account.
+     * @return The number processed with ILL or ERR verification.
+     */
+    public static String verificationIllErr(final String account) {
+        if (account.contains("?")) {
+            return account.concat(" ILL");
+        } else {
+            if (!checkSum(account)) {
+                return account.concat(" ERR");
+            }
+        }
+        return account;
```

Comment:
You may re-write this code, something like this.
```java
    public static String verificationIllErr(final String account) {
        if (account.contains("?")) {
            return account.concat(" ILL");
        }
        if (!checkSum(account)) {
            return account.concat(" ERR");
        }
        return account;
    }
```

Reply:
solved